### ISSUE TITLE
Make selection buffer work on X11 with firefox/chrome

### ISSUE THREAD

**Issue by jorisgio** [rust-lang/rust-playpen#69]
Hi,
on linux or BSD, with the X11 display server, the normal X11 selection buffer doesn't work in the textbox. You can't select the text in the textbox and middle click somewhere else to paste it. The clipboard selection buffer is working (copy/paste buttons from firefox), but it can be annoying because some applications (like terminal or vim) don't support this feature by default.

Would be amazing if it was fixed !

**Comment by zazdxscf** [comment]
**tl;dr:** I have `dom.event.clipboardevents.enabled` set to `false` in `about:config` (the default is true however), which prevents any kind of copy/pasting from occurring  but only within rust-playpen! (works fine on github, and even middle click pasting works (right after just selecting text), as long as: `clipboard.autocopy` is also true which it is by default )

Long read:
Oddly enough, for me the 'clipboard selection buffer' isn't working either, or rather it's working but seems to always copy `` and thus that's the only thing it pastes.
In other words, I've no way of copying out the code I see in the playpen, nor portions of it.

Sample: https://play.rust-lang.org/?code=fn%20main()%20%7B%0A%20%20%20%20fn%20diverges()%20-%3E%20!%20%7B%0A%20%20%20%20%20%20%20panic!(%22This%20function%20never%20returns!%22)%3B%0A%20%20%20%20%7D%0A%20%20%20%20let%20x%3A%20i32%20%3D%20diverges()%3B%0A%20%20%20%20let%20x%3A%20String%20%3D%20diverges()%3B%0A%20%20%20%20%0A%7D%0A&version=nightly

Also pasting into the playpen there, has no effect whatsoever. I cannot paste any new code (middle click or Ctrl+V or rightclick-menu->Paste).

Looks like it's because I have `dom.event.clipboardevents.enabled` set to `false` in `about:config` !! (although it works pretty much everywhere else, eg. github)
Yes, now everything works once I set it to true! (except, of course the original reported issue  which would most likely work if copy/pasting wouldn't depend on this `about:config` key being true)

**Comment by apoelstra** [comment]
This bug also affects me. Generally I have to find another website (e.g. codepad.org) to middle-click paste into, before using the mouse to Select All + Copy the text into the clipboard.

**Comment by aoaaxy** [comment]
The [above](https://github.com/rust-lang/rust-playpen/issues/69#issuecomment-155036257) is still true, for me, but only if I allow the scripts from `cdn.jsdelivr.net` (without which I don't have any formatting or coloring btw):

20:28:02			script	https://cdn.jsdelivr.net/ace/1.1.9/min/mode-rust.js
20:28:02			script	https://cdn.jsdelivr.net/ace/1.1.9/min/theme-github.js
20:28:01			script	https://cdn.jsdelivr.net/ace/1.1.9/min/ext-themelist.js
20:28:01			script	https://cdn.jsdelivr.net/ace/1.1.9/min/ace.js

I'm investigating "the issue" :)) I mean, enough is enough! hehe

PS: I don't know javascript/html/etc. so this might take a while...
